### ISSUE TITLE
nn.conv1d padding='same'

### ISSUE THREAD

**Issue by coder-sjn** [pytorch/pytorch#65510]
## ❓ Questions and Help

Hello, I just can’t figure out the way nn.Conv1d padding with padding='same'. Does pytorch now support this feature?

**Comment by H-Huang** [comment]
Docs (https://pytorch.org/docs/master/generated/torch.nn.Conv1d.html#conv1d) may need to be updated.

**Comment by sunnycasmir** [comment]
No, the padding='same' option for the nn.Conv1d layer is not supported by PyTorch at this time. Convolutional layers in PyTorch need an integer or a tuple of integers as the padding argument, which indicates how many zero-paddings to add to either side of the input.

Unlike other frameworks like TensorFlow, which provides the padding='same' option, PyTorch calculates the padding argument differently. To make sure that the output has the same spatial dimensions as the input, TensorFlow adds padding to the input and output independently. In PyTorch, this padding is applied to the input before the convolution process.

You must manually compute the padding values in order to have the same result as padding='same' in PyTorch. An illustration of how to accomplish this for nn.Conv1d is provided here:

You can make use of nn.functional.pad function from PyTorch to pad the input manually before passing it to the convulation layer.

import torch.nn.functional as F

# Example usage
batch_size = 1
in_channels = 1
in_size = 100
kernel_size = 5
stride = 1

input = torch.randn(batch_size, in_channels, in_size)
padding = get_same_padding(in_size, kernel_size, stride)
padded_input = F.pad(input, (padding[0], padding[1]))
conv = nn.Conv1d(in_channels, out_channels=1, kernel_size=kernel_size, stride=stride)
output = conv(padded_input)
print(output.shape)  # Output should be (1, 1, 100)
